constraint/pkg/client/drivers/interface.go

```diff
@@ -11,6 +11,10 @@ import (
 
 // A Driver implements Rego query execution of Templates and Constraints.
 type Driver interface {
+	// Name returns the name of the driver, used to uniquely identify a driver
+	// and in errors returned to the user.
+	Name() string
```

Comment:
No, we need Name() as a unique string identifier for a driver so that:

* Constraint templates can cite the driver name in their source code
* We can dereference that name into a specific driver instance
* We have a reliable comparator to see if a driver has changed.

Reply:
Because Driver is purely for internal consumption, if we were going to require extra data in order for driver to be useful, that is a de-facto breaking change.

If we wanted to add some kind of optional feature, it should not be in the interface, because then it would be required (typecasting can be used to convert to a more feature-rich interface if we want to support optional methods).